mage_ai/data_preparation/models/block/__init__.py

```diff
@@ -2074,6 +2075,24 @@ async def __get_outputs_async(
                 )
                 data_products.append(data)
                 continue
+            if type(data) is pl.DataFrame:
+                row_count, column_count = data.shape
```

Comment:
the row_count and column_count are sample data's row and column count.
if we want to get accurate values, we need store them.